### PR TITLE
Use containerregistry package to copy images

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: pkg/kotsadm/push_images.go [hunk 1]
```diff
@@ -164,57 +163,37 @@ func processImageTags(rootDir string, format string, imageName string, options t
...
 	}
+
 	if options.Registry.Username != "" && options.Registry.Password != "" {
-		destCtx.DockerAuthConfig = &containerstypes.DockerAuthConfig{
+		authConfig := authn.AuthConfig{
 			Username: options.Registry.Username,
 			Password: options.Registry.Password,
 		}
-	}
-	if os.Getenv("KOTSADM_INSECURE_SRCREGISTRY") == "true" {
```
Comment (divolgin) [inline pkg/kotsadm/push_images.go:190]: This never had any effect.  The flag is set unconditionally above.

File: pkg/image/builder.go [hunk 2]
```diff
@@ -616,11 +559,112 @@ func isUnauthorized(err error) bool {
...
+	runGC := false
+	defer func() {
+		if runGC {
+			runtime.GC()
+		}
+	}()
+
+	// "legacy" is an internal package and we can't import it
+	if desc.MediaType == containerregistrytypes.DockerManifestSchema1 || desc.MediaType == containerregistrytypes.DockerManifestSchema1Signed {
+		// srcAuth, err := authn.DefaultKeychain.Resolve(srcRef.Context().Registry)
```
Comment (salahalsaleh) [inline pkg/image/builder.go:588]: cleanup

File: pkg/image/builder.go [hunk 3]
```diff
@@ -616,11 +559,112 @@ func isUnauthorized(err error) bool {
...
+}
+
+func (r airgapImageOpener) Close() error {
+	if r.gzipReader != nil {
+		r.gzipReader.Close()
+	}
+	return r.fileReader.Close()
+}
+
+func RegistryImageImageFromReader(path string) (containerregistryv1.Image, error) {
```
Comment (salahalsaleh) [inline pkg/image/builder.go:635]: is this `ImageImage` intentional?

**Review by salahalsaleh — APPROVED** [review]

File: pkg/image/builder.go (hunk 3 above)
Comment (divolgin) [reply pkg/image/builder.go]: oops

**Pull request merged**